### PR TITLE
leetcode data crawl and store

### PR DESCRIPTION
**Pull request by JRakin**

### CODE REVIEW

File: pages/api/leetcode-api.ts
Comment (sifatul) [inline pages/api/leetcode-api.ts:6]: @JRakin better to trim data before parsing. also need to fail check for null or empty data

File: pages/api/platform/platform.ts
Comment (sifatul) [inline pages/api/platform/platform.ts:9]: @JRakin need to trim before parsing and also check for empty values

**Review by sifatul — CHANGES_REQUESTED** [review]

File: components/github-area.tsx
Comment (sifatul) [inline components/github-area.tsx:37]: @JRakin from next time please avoid changing files that you are not working with

File: components/leetcode-area.tsx
Comment (sifatul) [inline components/leetcode-area.tsx:50]: @JRakin need to update redux store

File: components/leetcode-area.tsx
Comment (sifatul) [inline components/leetcode-area.tsx:53]: @JRakin move this await too.

File: pages/api/platform/platform.ts
Comment (sifatul) [inline pages/api/platform/platform.ts:11]: @JRakin thanks for the check but what if param is empty?

**Review by sifatul — CHANGES_REQUESTED** [review]

File: components/github-area.tsx
Comment (sifatul) [inline components/github-area.tsx:64]: @JRakin before putting data we need to check whether its new or old data.

File: components/leetcode-area.tsx
Comment (sifatul) [inline components/leetcode-area.tsx:75]: @JRakin before putting data need to check wether its new or old data. if the data is new ( newly fetched from leetcode api) only then we need to save to reduce unnecessary call to our database

Comment (JRakin) [reply components/leetcode-area.tsx:75]: okay @sifatul vaiya I'll do that

Comment (JRakin) [reply components/leetcode-area.tsx:92]: @sifatul we not calling leetcode api when we have data in DB by that username

File: components/leetcode-area.tsx
Comment (sifatul) [inline components/leetcode-area.tsx:54]: @JRakin please write a bit cleaner code. 
initialize the json outside. then pass it to the `setLeetcodeUserInfo`

File: components/leetcode-area.tsx
Comment (sifatul) [reply components/leetcode-area.tsx:92]: @JRakin please check again. i think it will store data twice.
For example, your a userA is in database . 
1st render: this useEffect will be called and returned as redux does not have the data yet
2nd render: this use effect will be called again as we got data form database and updated redux:  leetcodeusername and profile url got  updated. This time it can pass the condition or not?

File: pages/api/leetcode-api.ts
Comment (sifatul) [inline pages/api/leetcode-api.ts:8]: @JRakin need to respond to !url or !variable

File: pages/api/platform/platform.ts
Comment (sifatul) [inline pages/api/platform/platform.ts:9]: @JRakin better to return something orelse debugging would be difficult

Comment (sifatul) [reply pages/api/platform/platform.ts:9]: @JRakin atleast a statuscode

**Review by sifatul — CHANGES_REQUESTED** [review]

File: pages/api/leetcode-api.ts
Comment (sifatul) [inline pages/api/leetcode-api.ts:19]: @JRakin could be covered in line 10 .  ` if (!url?.trim()) {`

**Review by sifatul — APPROVED** [review]
@JRakin feel free to merge after the small change

**Comment by JRakin** [comment]
@sifatul vaiya i can not merge as I'm not authorized I'm fixing this after that we can merge

**Pull request merged**
